scripts/build/TestImpactAnalysis/persistent_storage/tiaf_persistent_storage_local.py

```diff
@@ -70,3 +70,39 @@ def _retrieve_historic_data(self, config: dict, historic_workspace: str, histori
         except EnvironmentError as e:
             raise SystemError(
                 f"There was a problem the historic data file '{self._historic_data_file}': '{e}'.")
+
+    def _store_runtime_artifacts(self, runtime_artifact_dir : str):
+        """
+        Copy runtime artifacts from the provided directory to the historic storage directory.
+
+        @param runtime_artifact_dir: Path to runtime artifacts to copy.
+        """
+        source_directory = pathlib.Path(runtime_artifact_dir)
+        
+        storage_directory = self._historic_workspace.joinpath(pathlib.Path(self.RUNTIME_ARTIFACT_DIRECTORY))
+        storage_directory.mkdir(exist_ok=True, parents=True)
+
+        self._copy_files_from_directory_to_destination(source_directory, storage_directory)
+
+    def _store_coverage_artifacts(self, runtime_coverage_dir : str):
+        """
+        Copy runtime coverage artifacts from the provided directory to the historic storage directory.
+
+        @param runtime_coverage_dir: Path to the runtime coverage artifacts to copy.
+        """
+        source_directory = pathlib.Path(runtime_coverage_dir)
+
+        storage_directory = self._historic_workspace.joinpath(pathlib.Path(self.RUNTIME_COVERAGE_DIRECTORY))
+        storage_directory.mkdir(exist_ok=True, parents=True)
+
+        self._copy_files_from_directory_to_destination(source_directory, storage_directory)
+
+    def _copy_files_from_directory_to_destination(self, source_directory: pathlib.Path, target_directory : pathlib.Path):
+        """
+        Copies all files in source directory to the target directory.
+
+        @param source_directory: pathlib.Path to directory to copy files from.
+        @param target_direcotry: pathlib.Path to directory to store files in.
+        """
+        for artifact_path in source_directory.iterdir():
+            shutil.copy2(artifact_path, target_directory.joinpath(artifact_path.name))
```

Comment:
Might want to put some error handling here as many issues can occur here (permissions, disk space, etc.)

Reply:
Good point, will add in a try except